Fix the following warnings:

  components/history/core/browser/history_types.h:73:45: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr std::optional<std::string> kNoAppIdFilter = std::nullopt;
                                              ^

--- a/components/history/core/browser/history_types.h
+++ b/components/history/core/browser/history_types.h
@@ -70,6 +70,7 @@ constexpr VisitedLinkID kInvalidVisitedL
 typedef std::map<VisitID, VisitSource> VisitSourceMap;
 
 // Constant used to represent that no app_id is used for matching.
+[[clang::no_destroy]]
 inline constexpr std::optional<std::string> kNoAppIdFilter = std::nullopt;
 
 // VisitRow -------------------------------------------------------------------
